indexer-app/src/cmr/indexer/data/concept_parser.clj

```diff
@@ -57,11 +58,26 @@
   [context concept]
   (edn/read-string (:metadata concept)))
 
-(doseq [concept-type (concepts/get-generic-concept-types-array)]
+(doseq [concept-type concepts/get-generic-non-draft-concept-types-array]
   (defmethod parse-concept concept-type
     [context concept]
     (json/parse-string (:metadata concept) true)))
 
+(doseq [concept-type concepts/get-draft-concept-types-array]
+  (defmethod parse-concept concept-type
+    [context concept]
+    ;; If the draft record is a json record, then parse it,
+    ;; otherwise figure out what the draft record concept type is
+    ;; and let the parsing code parse it to umm-c.
+    (if (string/includes? (:format concept) "json")
+      (json/parse-string (:metadata concept) true)
+      (let [concept-type (:concept-type concept)]
+        (if (concepts/is-draft-concept? concept-type)
```

Comment:
Should we move this `(if (concepts/is-draft-concept? concept-type)` to the top of the function before the `(if (string/includes? (:format concept) "json")` check?

Reply:
Drafts can have both json or xml records, while right now non drafts are just json, but that can change.  Regardless, if I change the if statement, then I will need to add an extra if to have the same logic.

(if (concepts/is-draft-concept? concept-type)
    (if (string/includes? (:format concept) "json")
      (json/parse-string (:metadata concept) true)
      (let [draft-concept-type (concepts/get-concept-type-of-draft concept-type)
            concept (assoc concept :concept-type draft-concept-type)]
        (umm/parse-metadata context concept)))
    (if (string/includes? (:format concept) "json")
      (json/parse-string (:metadata concept) true)
      (umm/parse-metadata context concept)))

I think it is simpler leaving it as is.